test/core/shared-everything-threads/shared-everything-threads/tables.wast

```diff
@@ -0,0 +1,206 @@
+;; Check the `shared` attribute on tables.
+
+(module
+  ;; Imported.
+  (table (import "spectest" "table_ref") shared 1 (ref null (shared func)))
+  (table (import "spectest" "table_ref_with_max") shared 1 1 (ref null (shared func)))
+
+  ;; Normal.
+  (table shared 1 (ref null (shared func)))
+  (table shared 1 1 (ref null (shared func)))
+
+  ;; Inlined.
+  (table shared (ref null (shared func)) (elem (ref.null (shared func))))
+)
+
+;; Note that shared elements can live within an unshared table.
```

Comment:
It would be good to test that the reverse is invalid. Also, we can test the interaction with shared and unshared globals used in the table initializers.

Reply:
> It would be good to test that the reverse is invalid.

We do; they're just a few lines down. I'll move those closer to this assertion.

> Also, we can test the interaction with shared and unshared globals used in the table initializers.

Were you thinking of variations on this kind of thing?

```wat
(module
  (type $f (shared (func)))
  (global $g (shared (ref null $f)) (ref.null $f))
  (table $t shared 1 (ref null $f) (ref.null $f))
  (elem (global.get $g))
)
```

Reply:
Yes, although that test could be simplified by removing the table entirely and just keeping the element segment.

Reply:
Really? Both `wasm-tools` and `wabt` seem to expect an `elem` to have a preceding `table` definition or we get an error. E.g., `(module (elem (i32.const 0) funcref))` is invalid in `wat2wasm` v1.0.33.

Reply:
...but see 740c560 (a lot more notation is necessary than I might have expected to convince `wasm-tools` it is valid).

Reply:
> Really? Both `wasm-tools` and `wabt` seem to expect an `elem` to have a preceding `table` definition or we get an error. E.g., `(module (elem (i32.const 0) funcref))` is invalid in `wat2wasm` v1.0.33.

That module is invalid because it defines an active segment, which needs a table to be initialized into. Get rid of the `(i32.const 0)` offset and it should validate.